packages/protocol/test/bridge/TokenVault.test.ts

```diff
@@ -190,4 +231,183 @@ describe("TokenVault", function () {
+        });
+
+        it("should throw if isBridgedToken, and canonicalToken.addr == address(0)", async function () {
+            const toBytes32 = (bn: BigNumber) => {
+                return ethers.utils.hexlify(
+                    ethers.utils.zeroPad(bn.toHexString(), 32)
+                );
+            };
+
+            const isBridgedTokenSlot = await getSlot(
+                hre,
+                bridgedToken.address,
+                201
+            );
+            await helpers.setStorageAt(
+                L1TokenVault.address,
+                isBridgedTokenSlot,
+                1
+            );
+            const bridgedToCanonicalSlot = await getSlot(
+                hre,
+                bridgedToken.address,
+                202
+            );
+            await helpers.setStorageAt(
+                L1TokenVault.address,
+                bridgedToCanonicalSlot,
+                ethers.constants.AddressZero
+            );
+
+            await helpers.setStorageAt(bridgedToken.address, 203, 1000000);
+            await bridgedToken.approve(owner.address, 1000);
+            const accountBalanceSlot = await getSlot(hre, owner.address, 201);
+            await helpers.setStorageAt(
+                bridgedToken.address,
+                accountBalanceSlot,
+                toBytes32(ethers.BigNumber.from("10")).toString()
+            );
+
+            await expect(
+                L1TokenVault.connect(owner).sendERC20(
+                    destChainId,
+                    nonOwner.address,
+                    bridgedToken.address,
+                    1,
+                    20000000,
+                    1000,
+                    owner.address,
+                    "",
+                    {
+                        value: 1,
+                    }
+                )
+            ).to.be.revertedWith("V:canonicalToken");
+        });
+
+        it("should pass and emit ERC20Sent Event", async function () {
+            const mockTokenVaultFactory = await smock.mock<TokenVault__factory>(
+                "TokenVault"
+            );
+            const L1TokenVault = await mockTokenVaultFactory.deploy();
+            await L1TokenVault.init(tokenVaultAddressManager.address);
+            await tokenVaultAddressManager.setAddress(
+                `${hre.network.config.chainId}.token_vault`,
+                L1TokenVault.address
+            );
+
+            const toBytes32 = (bn: BigNumber) => {
+                return ethers.utils.hexlify(
+                    ethers.utils.zeroPad(bn.toHexString(), 32)
+                );
+            };
+
+            await L1TokenVault.setVariable("isBridgedToken", {
+                [bridgedToken.address]: true,
+            });
+
+            await L1TokenVault.setVariable("bridgedToCanonical", {
+                [bridgedToken.address]: weth,
+            });
+
+            await helpers.setStorageAt(bridgedToken.address, 203, 1000000);
```

Comment:
If `smock.setVariable` works, i think we should change all the `getSlot` and `setStorageAt` to `smock.setVariable` in all tests? these `setStorageAt` with a hardcode slot number is really non-trivial to maintain, eveytime anyone changes the contract's code, u may need to update this `203` or other mapping slot to a new value manually.

Reply:
I can go through and change that, should I make a separate PR or do it for all tests using setStorageAt on this branch?

Reply:
Lets make a separate PR that targets to this PR?